sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/EqualPartitionsBalancingStrategy.java

```diff
@@ -24,6 +24,7 @@
     private final int minPartitionCount;
     private final int maxPartitionCount;
     private final Duration leaseExpirationInterval;
+    private volatile int countAssignedLeases;
```

Comment:
why we need volatile here, two+ threads can read/change it ?

Reply:
@milismsft why don't we use AtomicInteger here?

Reply:
Yes, though not very likely; the parent call is in a flatMap but should only be triggered on one thread at the time. I've added the volatile since the current value is dynamically changed over the application execution.